components/event-sources/upgrade-job/process/convert_triggers_to_kyma_subs.go

```diff
@@ -0,0 +1,125 @@
+package process
+
+import (
+	"fmt"
+
+	kymaeventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	kymaeventingapp "github.com/kyma-project/kyma/components/eventing-controller/pkg/application"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+)
+
+var _ Step = &ConvertTriggersToKymaSubscriptions{}
+
+const (
+	eventVersionKey = "eventtypeversion"
+	eventSourceKey  = "source"
+	eventTypeKey    = "type"
+)
+
+type ConvertTriggersToKymaSubscriptions struct {
+	name    string
+	process *Process
+}
+
+func NewConvertTriggersToKymaSubscriptions(p *Process) ConvertTriggersToKymaSubscriptions {
+	return ConvertTriggersToKymaSubscriptions{
+		name:    "Convert Triggers to Kyma Subscriptions",
+		process: p,
+	}
+}
+
+func (s ConvertTriggersToKymaSubscriptions) Do() error {
+	for _, trigger := range s.process.State.Triggers.Items {
+		// Generate equivalent subscription for this trigger
+		sub := s.NewSubscription(&trigger)
+
+		_, err := s.process.Clients.KymaSubscription.Create(sub)
+		if err != nil {
```

Comment:
Does it make sense to not return an error if `k8serrors.IsAlreadyExists(err)` is true? 

Because this means that we already migrated this trigger before and we don't need to return an error. Or maybe delete and recreate again instead of skipping.

Reply:
Fixed